be/CMakeLists.txt

```diff
@@ -612,7 +612,16 @@ FUNCTION(ADD_BE_LIB LIB_NAME)
     set(ALL_BE_LIBS "${ALL_BE_LIBS};${LIB_NAME}" CACHE INTERNAL "All BE libraries")
 ENDFUNCTION()
 
-set (STARROCKS_STATIC_LINK_LIBS ${WL_LINK_STATIC} -lgomp -lbfd)
+# Check if libsframe exists (needed for binutils 2.40+ on newer distros like Ubuntu 24.04)
+find_library(SFRAME_LIB sframe)
```

Comment:
considering if there is a way to check the bfd version, and conditionally link the libsframe only when the version greater than x.y.

or if the `pkg-config` can do the tricky, using pkg-config to get the link dependency of libbfd, and add it together with `-lbfd`